android/KMEA/app/src/main/java/com/tavultesoft/kmea/util/FileUtils.java

```diff
@@ -278,21 +277,16 @@ public static boolean isKeymanLink(String u) {
       return ret;
     }
     String lowerU = u.toLowerCase();
-    Pattern pattern = Pattern.compile("^keyman:(\\w+)\\?(.+)");
+    String patternFormatStr = String.format("^(%s|%s)/keyboards/install/(\\w+)(\\?bcp47=)?(.+)?",
```

Comment:
This pattern has been duplicated...

Reply:
The regex originally was here:
1. to determine whether a URL was a keyboard download link
2. so I could verify my regex was correct in FileUtilsTest.

Now that I need to match components from the regex (beyond a boolean return), I likely need to refactor this

Reply:
Refactored to `KMPLink`